packages/renderer/src/lib/onboarding/OnboardingItem.svelte

```diff
@@ -0,0 +1,93 @@
+<script lang="ts">
+import { onDestroy, onMount } from 'svelte';
+import type { OnboardingStepItem } from '../../../../main/src/plugin/api/onboarding';
+import Markdown from '../markdown/Markdown.svelte';
+import type { ContextUI } from '../context/context';
+import { SCOPE_ONBOARDING } from './onboarding-utils';
+export let extension: string;
+export let item: OnboardingStepItem;
+export let getContext: () => ContextUI;
+export let executeCommand: (command: string) => Promise<void>;
+
+const onboardingContextRegex = new RegExp(/\${onboardingContext:(.+?)}/g);
+const globalContextRegex = new RegExp(/\${onContext:(.+?)}/g);
+let html;
+let isMarkdown = false;
+$: buttons = new Map<string, string>();
+const eventListeners = [];
+onMount(() => {
+  const itemHtml = createItem(item);
+  html = itemHtml;
+  const clickListener = async e => {
+    if (e.target instanceof HTMLButtonElement) {
+      const buttonId = e.target.id;
+      let command = buttons.get(buttonId);
+      if (command) {
+        await executeCommand(command);
+      }
+    }
+  };
+  eventListeners.push(clickListener);
+  /* eslint-disable @typescript-eslint/no-misused-promises */
+  document.addEventListener('click', clickListener);
+});
+onDestroy(() => {
+  eventListeners.forEach(listener => document.removeEventListener('click', listener));
+});
+function createItem(item: OnboardingStepItem): string {
+  let html = '';
+  switch (item.component) {
+    case 'button': {
+      const buttonId = `button-${buttons.size}`;
+      const value = `<button id="${buttonId}" class="bg-purple-700 py-2 px-2.5 rounded-md">${item.label}</button>`;
+      buttons.set(buttonId, item.command);
+      buttons = buttons;
+      html = value;
+      break;
+    }
+    default: {
+      html = replacePlaceholders(item.value);
+      isMarkdown = true;
+    }
+  }
+  return html;
+}
+
+function replacePlaceholders(label: string): string {
+  let newLabel = label;
+  newLabel = replacePlaceHoldersRegex(onboardingContextRegex, newLabel, `${extension}.${SCOPE_ONBOARDING}`);
+  newLabel = replacePlaceHoldersRegex(globalContextRegex, newLabel);
+  return newLabel;
+}
+
+function replacePlaceHoldersRegex(regex: RegExp, label: string, prefix?: string) {
+  let arr;
+  // eslint-disable-next-line eqeqeq
```

Comment:
why the rule is being disabled ?

Reply:
bc it should be `!== null` but null values are not permitted, so i replace with `!= undefined` and it complained again with the eqeqeq eslint rule

Reply:
`!== undefined` ? (two == sign)

Reply:
use the more recent matchAll ?
```
str.matchAll(regexp);
```